nbresuse/__init__.py

```diff
@@ -178,19 +97,20 @@ def _mem_limit_default(self):
         Note that this does not actually limit the user's CPU usage!
 
         Defaults to reading from the `CPU_LIMIT` environment variable. If
-        set to 0, no CPU usage limit is displayed.
+        set to 0, the total CPU count available is displayed.
         """
     ).tag(config=True)
 
     @default('cpu_limit')
     def _cpu_limit_default(self):
         return float(os.environ.get('CPU_LIMIT', 0))
 
+
 def load_jupyter_server_extension(nbapp):
     """
     Called during notebook start
     """
     resuseconfig = ResourceUseDisplay(parent=nbapp)
     nbapp.web_app.settings['nbresuse_display_config'] = resuseconfig
-    route_pattern = url_path_join(nbapp.web_app.settings['base_url'], '/metrics')
-    nbapp.web_app.add_handlers('.*', [(route_pattern, MetricsHandler)])
+    callback = ioloop.PeriodicCallback(PrometheusHandler(nbapp), 1000)
```

Comment:
The name PrometheusHandler does not mean its a API handler, it is a bit misleading. The PrometheusHandler class is simply a functor, callable/runnable, and ioloop.PeriodicCallback simply runs the __call__ function every 1000 millis, in order to update the value in the prometheus metrics. Exposing the metrics is done by jupyter itself, which is running a prometheus server in its core, the code simply uses the existing infrastructure.

Reply:
Oh, oh wow, OK yes that sounds like a very good solution actually.

I'm starting to see the light now I think on using it exclusively, although I should double check with Yuvi and others before making any final decision about how to structure future releases.

But this would probably make NBResuse much easier to merge into JupyterLab. I think at this point most of my reluctance to merge this was mostly because of my lack of knowledge about Prometheus. But even if there are any temporary hiccups, it won't affect any previous PyPI releases, and people who install from source via GitHub know that they're getting the bleeding edge and possibly unstable code.